views/kanban-view/src/components/Board/Board.tsx

```
@@ -130,92 +54,195 @@ export default function Board({ perspective, source, agent }: BoardProps) {
+      setNamedOptions(namedOpts);
+      Object.keys(namedOpts).forEach((key, index) => {
+        if (index === 0) setSelectedProperty(key);
+      });
+    });
+  }
+
+  function addTaskToColumn(columns, task, propertyName) {
+    return Object.keys(columns).reduce((acc, key) => {
+      const column = columns[key];
+      return {
+        ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)